rustls/src/msgs/deframer.rs

```diff
@@ -87,10 +91,22 @@ impl MessageDeframer {
                 }
             };
 
-            // If we're in the middle of joining a handshake payload and the next message is not of
-            // type handshake, yield an error. Return CCS messages immediately without decrypting.
+            // Return CCS messages and early plaintext alerts immediately without decrypting.
             let end = start + rd.used();
-            if m.typ == ContentType::ChangeCipherSpec && self.joining_hs.is_none() {
+            let type_is_ccs = m.typ == ContentType::ChangeCipherSpec;
+            let has_decrypted = record_layer.has_decrypted();
+            let type_is_alert = m.typ == ContentType::Alert;
+            let payload_is_alert = m.payload.0.len() <= 2;
```

Comment:
perhaps this could be recast as something like:

```rust
let allowed_plaintext = match m.typ {
  ContentType::ChangeCipherSpec => true,
  ContentType::Alert if version_is_tls13 && !record_layer.has_decrypted && m.payload.0.len() <= 2 => true,
  _ => false,
};
(...)
if self.joining_hs.is_none() && allowed_plaintext {
  (...)
}
```

Reply:
Yup, I like that better than my previous suggestion.